lib/tapioca/generators/base_generator.rb

```diff
@@ -0,0 +1,110 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "active_support/all"
+
+module Tapioca
+  module Generators
+    class BaseGenerator
+      extend(T::Sig)
+      extend(T::Helpers)
+
+      abstract!
+
+      sig { returns(Config) }
+      attr_reader :config
+
+      sig { params(config: Config).void }
```

Comment:
`Config` seems to be a concept closely related to a CLI tool. Why not decoupling this and passing all the config options as keyword parameter?

This will have at least two benefits:
1. Better separation of concerns (since a generator will be unable to access config value unrelated to its purpose)
2. Simplify the way we test the generators (no need to create a config object)

Reply:
I was doing this initially to maintain compatability with the existing implementation. But that was the game plan for future changes, as IIRC, `Config` will be going through some changes as well.